sdk/keyvault/azure-security-keyvault/azure/security/keyvault/keys/_client.py

```diff
@@ -3,266 +3,715 @@
-        key = self._deserialize("Key", response)
+        if version is None:
+            version = ""
 
-        return key
+        bundle = self._client.get_key(self.vault_url, name, version, error_map={404: ResourceNotFoundError})
+        return Key._from_key_bundle(bundle)
 
     def get_deleted_key(self, name, **kwargs):
-        # type: (str, Any) -> DeletedKey
-
-        url = "/".join([self.vault_url, "deletedkeys", name])
-
-        request = HttpRequest("GET", url)
-        request.format_parameters({"api-version": self.API_VERSION})
-        response = self._pipeline.run(request, **kwargs).http_response
-        if response.status_code != 200:
-            raise HttpResponseError("Request failed with code {}: '{}'".format(response.status_code, response.text()))
-        deleted_key = self._deserialize("DeletedKey", response)
-
-        return deleted_key
-
-    def get_all_deleted_keys(self, max_page_size=None, **kwargs):
-        # type: (Optional[int], Any) -> DeletedKeyItemPaged
-        url = "{}/{}".format(self.vault_url, "deletedkeys")
-        paging = functools.partial(self._internal_paging, url, max_page_size)
-        return DeletedKeyItemPaged(paging, self._deserialize.dependencies)
-
-    def get_all_keys(self, max_page_size=None, **kwargs):
-        # type: (Optional[int], Any) -> KeyItemPaged
-        url = "{}/{}".format(self.vault_url, "keys")
-        paging = functools.partial(self._internal_paging, url, max_page_size)
-        return KeyItemPaged(paging, self._deserialize.dependencies)
-
-    def get_key_versions(self, name, max_page_size=None, **kwargs):
-        # type: (Optional[int], Any) -> KeyItemPaged
-        url = "/".join([self.vault_url, "keys", name, "versions"])
-        paging = functools.partial(self._internal_paging, url, max_page_size)
-        return KeyItemPaged(paging, self._deserialize.dependencies)
+        # type: (str, Mapping[str, Any]) -> DeletedKey
+        """Gets the public part of a deleted key.
+
+        The Get Deleted Key operation is applicable for soft-delete enabled
+        vaults. While the operation can be invoked on any vault, it will return
+        an error if invoked on a non soft-delete enabled vault. This operation
+        requires the keys/get permission.
+
+        :param name: The name of the key.
+        :type name
+        :returns: The deleted key
+        :rtype: ~azure.security.keyvault.keys._models.DeletedKey
+
+        Example:
+            .. literalinclude:: ../tests/test_examples_keys.py
+                :start-after: [START get_deleted_key]
+                :end-before: [END get_deleted_key]
+                :language: python
+                :dedent: 4
+                :caption: Retrieves a deleted key from the key vault
+        """
+        bundle = self._client.get_deleted_key(self.vault_url, name, error_map={404: ResourceNotFoundError})
+        return DeletedKey._from_deleted_key_bundle(bundle)
+
+    def list_deleted_keys(self, **kwargs):
```

Comment:
My understanding from the API review meeting was that for all languages this should be get_deleted_keys, instead of using `list`. Is this not applying to Python?

Reply:
This was one of the TODO items brought up in ADP Board meeting.  I have opened an issue to discuss this change. For now, we will be going ahead with calling it `list` and updating it later.